auth/provider/types.go

```diff
@@ -0,0 +1,39 @@
+package provider
+
+import (
+	"strings"
+	"time"
+)
+
+type Profile struct {
+	ID        int       `json:"id"`
+	Login     string    `json:"login"`
+	Name      string    `json:"name"`
+	Email     string    `json:"email"`
+	TwoFactor bool      `json:"two_factor"`
+	CreatedAt time.Time `json:"created_at"`
+}
+
+type Provider interface {
+	GetProfile(accessToken string) (*Profile, error)
+}
+
+var supportedProviders = []string{
```

Comment:
Much less complexity. Are we going to be putting `github` / `gitlab` into consts?

Reply:
Seems that you did already, can you make use of them here please?

Reply:
We are in `auth/handlers`, but here I thought it's the only place. I can copy code from `handlers` here too